### PR TITLE
T-93 Category API - Dashboard

### PR DESCRIPTION
**Pull request by VL-CZ**
# Category API for dashboard 
API contains several views:
- `CategoryViewDashboard`: list all categories & add new ones
- `CategoryDetailViewDashboard`: get category by ID, update, delete
- `CategoryChildrenViewDashboard`: get category by ID including its children

Authorization not implemented yet

### CODE REVIEW

**Review by lakatop — COMMENTED** [review]
Can we get rid of authentication TODOs ?

**Comment by VL-CZ** [comment]
> Can we get rid of authentication TODOs ?

Technically yes, but for now, I'd leave it there (to not forget about it).

**Comment by VL-CZ** [comment]
Actually, I'd rather keep it this way for now. We can implement it later (maybe we can do all of the authentication at once), but for now it would be IMO easier to test it / call from frontend with no auth.

**Comment by michalpulpan** [comment]
> Actually, I'd rather keep it this way for now. We can implement it later (maybe we can do all of the authentication at once), but for now it would be IMO easier to test it / call from frontend with no auth.

Agree, let's choose easy route first so that we have at least something 😄

File: src/backend/core/product/migrations/0017_alter_productvariantimage_unique_together_and_more.py
[the remark this answers is not in the document]
Comment (michalpulpan) [reply src/backend/core/product/migrations/0017_alter_productvariantimage_unique_together_and_more.py:1]: Yeah, I think this should be fine. On my #T-55 I'll definitely have conflict, but considering our database (without data), we can always delete all migration files and database and start from scratch (perform initial migration).

**Comment by lakatop** [comment]
Alright then, lets leave it like that for now.  ✅

**Review by lakatop — APPROVED** [review]

**Pull request merged**